src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -257,12 +253,12 @@ void BaseCouplingScheme::advance()
   }
 }
 
-void BaseCouplingScheme::storeDataInWaveforms()
+void BaseCouplingScheme::storeExtrapolationData()
 {
   PRECICE_TRACE(_timeWindows);
   for (DataMap::value_type &pair : _allData) {
     PRECICE_DEBUG("Store data: {}", pair.first);
-    _waveforms[pair.first]->store(pair.second->values());
+    pair.second->storeDataInExtrapolation();
```

Comment:
```suggestion
    pair.second->storeExtrapolationData();
```
?
If we really use two different names for the two different functions, the name should explain the difference.